program/src/instruction.rs

```diff
@@ -402,8 +406,74 @@ pub enum StakeInstruction {
     DistributeRewards(u64),
 }
 
+impl StakeInstruction {
+    /// Unpacks a byte buffer into a
+    /// [PaladinRewardsInstruction](enum.PaladinRewardsInstruction.html).
+    pub fn unpack(input: &[u8]) -> Result<Self, ProgramError> {
+        match input.split_first() {
+            // 0 - InitializeConfig: u64 (8) + u16 (2)
+            Some((&0, rest)) if rest.len() == 10 => Ok(StakeInstruction::InitializeConfig {
+                cooldown_time_seconds: i64::from_le_bytes(*array_ref![rest, 0, 8]),
```

Comment:
Nice, so `array_ref` here gets you around that annoying `.try_into().unwrap()` for `[u8; 8]`?

Reply:
Yes, exactly.